ground/gcs/src/libs/tlmapcontrol/core/providerstrings.cpp

```diff
@@ -67,13 +68,31 @@ QString ProviderStrings::decrypt(QString str)
         ++pos;
     }
 
-    return QString::fromUtf8(qUncompress(ba));
+    QString decryptedString = QString::fromUtf8(qUncompress(ba));
+    if (decryptedString.isEmpty()) {
+        throw std::domain_error(QString("Decrypted to an empty vector!. Input vector: ").append(inputString).toStdString());

```

Comment:
What happens when this is thrown? What catches it and is it shown to the user somehow?